File: 844-backspace-string-compare/844-backspace-string-compare.cpp
class Solution {
public:
    stack<char> helper(string x)
    {
        stack<char> st;
        for(char c : x)
        {
            if(c!= '#') st.push(c);
            else if(st.size()) st.pop();
        }
        return st;
    }
    bool backspaceCompare(string s, string t) {
        return helper(s) == helper(t);
    }
};

File: 844-backspace-string-compare/844-backspace-string-compare.py
class Solution:
    def helper(self,x : str)->list:
        st = []
        for i in x:
            if i != '#':
                st.append(i)
            elif len(st):
                st.pop()
        return st
    def backspaceCompare(self, s: str, t: str) -> bool:
        return self.helper(s) == self.helper(t)